src/datoso_plugin_internetarchive/fetch.py

```diff
@@ -45,7 +46,7 @@ def print_progress(done):
 
     with ThreadPoolExecutor(max_workers=int(config.get('DOWNLOAD', 'Workers', fallback=10))) as executor:
         futures = [
-            executor.submit(download_dat, os.path.join(ia.get_download_path(), file['name'])) for file in dats
+            executor.submit(download_dat, urljoin(urljoin(ia.get_download_path(), archive.dat_folder), file['name'])) for file in dats
```

Comment:
```suggestion
            executor.submit(download_dat, RequestUtils.urljoin(ia.get_download_path(), file['name'])) for file in dats
```

Reply:
Looks like I didn't realise the `file['name']` value already included the `DATs/` folder path, and my tweak only worked through blind luck; in that when using the urljoin lib, it will discard anything after the final slash, and replace it with the second parameter. The first join did the task of adding `DATs` (no slash) to the end of the `get_download_path()` result, making it something like `https://...En-ROMs/DATs`, which when urljoined with a `file['name']` value, REPLACED the ending `DATs` with the `DATs/(filenamehere)`.